packages/experimental/Badge/src/CounterBadge/CounterBadge.tsx

```diff
@@ -0,0 +1,42 @@
+/** @jsx withSlots */
+import { Children, ReactNode } from 'react';
+import { View } from 'react-native';
+import { counterBadgeName, CounterBadgeType, CounterBadgeProps } from './CounterBadge.types';
+import { compose, withSlots, mergeProps, UseSlots } from '@fluentui-react-native/framework';
+import { createIconProps } from '@fluentui-react-native/interactive-hooks';
+import { stylingSettings } from './CounterBadge.styling';
+import { Icon } from '@fluentui-react-native/icon';
+import { badgeLookup } from '../Badge';
+import { useCounterBadge } from './useCounterBadge';
+import { TextV1 as Text } from '@fluentui-react-native/text';
+
+export const CounterBadge = compose<CounterBadgeType>({
+  displayName: counterBadgeName,
+  ...stylingSettings,
+  slots: {
+    root: View,
+    icon: Icon,
+    text: Text,
+  },
+  useRender: (userProps: CounterBadgeProps, useSlots: UseSlots<CounterBadgeType>) => {
+    const iconProps = createIconProps(userProps.icon);
+    const badge = useCounterBadge(userProps);
+
+    const Slots = useSlots(badge.props, (layer) => badgeLookup(layer, badge.props));
+
+    return (final: CounterBadgeProps, ...children: ReactNode[]) => {
+      const { icon, iconPosition = 'before', ...mergedProps } = mergeProps(badge.props, final);
+      const { displayCount, showBadge } = badge.state;
+      return (
+        showBadge && (
+          <Slots.root {...mergedProps}>
+            {icon && iconPosition === 'before' && <Slots.icon {...iconProps} />}
+            <Slots.text>{displayCount}</Slots.text>
```

Comment:
nit: This means that any "count" passed in via an inner render function from a parent component is ignored.  Not necessarily a big deal, just that this would be the result of how things are coded.

Reply:
Maybe it would be better to move the displayCount logic here, so counter value from inner function would, also, work